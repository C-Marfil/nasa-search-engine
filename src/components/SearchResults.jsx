import React from "react";
import { useParams, Link } from "react-router-dom";
import PropTypes from "prop-types";
import "../styles/search-results.css";

const SearchResults = ({ results, setLevel, level }) => {
  const { q } = useParams();

  return (
    <>
      {level === "shallow" && <p>Search Results</p>}
      <div className="images-container">
        {results.map((e) => {
          return (
            <Link
              key={e}
              className="image"
              to={`/${q}/image-detail`}
              onClick={() => setLevel("shallow")}
              state={{ src: `${e}` }}
            >
              <img key={e} className="image" src={e} alt="space-is-pretty" />
            </Link>
          );
        })}
      </div>
    </>
  );
};

SearchResults.propTypes = {
  results: PropTypes.arrayOf(PropTypes.string).isRequired,
  level: PropTypes.string.isRequired,
  setLevel: PropTypes.func.isRequired,
};

export default SearchResults;
