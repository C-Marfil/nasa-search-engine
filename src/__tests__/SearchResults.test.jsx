import { fireEvent, React, render, screen } from "@testing-library/react";
import { MemoryRouter as Router } from "react-router-dom";
import { createBrowserHistory } from "history";
import SearchResults from "../components/SearchResults";

describe("Search Results component", () => {
  const validProps = {
    results: ["http//1", "http//2u"],
    setLevel: jest.fn(),
    level: "shallow",
  };

  it("Snapshot test", () => {
    const { asFragment } = render(
      <SearchResults
        results={validProps.results}
        setLevel={validProps.setLevel}
        level={validProps.level}
      />,
      { wrapper: Router }
    );

    expect(asFragment()).toMatchSnapshot();
  });

  it("renders elements correctly", () => {
    render(
      <Router initialEntries={["/moon"]}>
        <SearchResults
          results={validProps.results}
          setLevel={validProps.setLevel}
          level={validProps.level}
        />
      </Router>
    );

    expect(screen.getByText("Search Results")).toBeInTheDocument();
    expect(screen.getAllByRole("img")).toHaveLength(2);
  });

  // I've tried...
  xit("Tests the img's Link elements", async () => {
    const history = createBrowserHistory({ initialEntries: ["/moon"] });
    render(
      <Router history={history}>
        <SearchResults
          results={validProps.results}
          setLevel={validProps.setLevel}
          level={validProps.level}
        />
      </Router>
    );

    expect(history.location.pathname).toBe("/moon");
    await fireEvent.click(screen.getByAltText("space-is-pretty"));
    expect(history.location.pathname).toBe("/");
  });
});
