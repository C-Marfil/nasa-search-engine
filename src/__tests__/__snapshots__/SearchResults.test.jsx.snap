// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search Results component Snapshot test 1`] = `
<DocumentFragment>
  <p>
    Search Results
  </p>
  <div
    class="images-container"
  >
    <a
      class="image"
      href="/undefined/image-detail"
    >
      <img
        alt="space-is-pretty"
        class="image"
        src="http//1"
      />
    </a>
    <a
      class="image"
      href="/undefined/image-detail"
    >
      <img
        alt="space-is-pretty"
        class="image"
        src="http//2u"
      />
    </a>
  </div>
</DocumentFragment>
`;
